File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/shop/statistics/shopStatistics.ts

export interface ShopStatistics {
    inCartCopies: number;
    inOrderCopies: number;
    soldCopies: number;
    canceledCopies: number;
    orderAmount: number;
    canceledOrderAmount: number;
    averagePrice: number;
    earnedMoney: number;
    orderAmountInDays: Map<Date, number>;
}

export function getDefaultShopStatistics(): ShopStatistics {
    return {
        inCartCopies: 0,
        inOrderCopies: 0,
        soldCopies: 0,
        canceledCopies: 0,
        orderAmount: 0,
        canceledOrderAmount: 0,
        averagePrice: 0,
        earnedMoney: 0,
        orderAmountInDays: new Map<Date, number>()
    }
}

File: src/ELibrary.Backend/ShopApi/Features/StatisticsFeature/Domain/Models/ShopStatistics.cs
namespace ShopApi.Features.StatisticsFeature.Domain.Models
{
    public class ShopStatistics
    {
        public required long InCartCopies { get; set; }
        public required long InOrderCopies { get; set; }
        public required long SoldCopies { get; set; }
        public required long CanceledCopies { get; set; }
        public required long OrderAmount { get; set; }
        public required long CanceledOrderAmount { get; set; }
        public required decimal AveragePrice { get; set; }
        public required decimal EarnedMoney { get; set; }
        public Dictionary<DateTime, long> OrderAmountInDays { get; set; } = new();
    }
}
